indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/LagBasedAutoScalerConfig.java

```diff
@@ -61,7 +62,8 @@ public LagBasedAutoScalerConfig(
           @Nullable @JsonProperty("scaleInStep") Integer scaleInStep,
           @Nullable @JsonProperty("scaleOutStep") Integer scaleOutStep,
           @Nullable @JsonProperty("enableTaskAutoScaler") Boolean enableTaskAutoScaler,
-          @Nullable @JsonProperty("minTriggerScaleActionFrequencyMillis") Long minTriggerScaleActionFrequencyMillis
+          @Nullable @JsonProperty("minTriggerScaleActionFrequencyMillis") Long minTriggerScaleActionFrequencyMillis,
+          @Nullable @JsonProperty("useDefaultTotalLag") Boolean useDefaultTotalLag
```

Comment:
I think it would be better to have this be an enum, like what was done in the initial patch. Then the docs for this field would tell the user to set this to one of `TOTAL`, `MAX` or `AVG` with the default implementation using the total lag.

@kfaraz do you have any concerns with this approach since you had some comments about the use of an enum in an earlier patch.

On naming - I think the name could be something like `lagStatsType` The word `default` in the current name is confusing to me. Once there is agreement on the approach, there should also be a doc update in the `supervisor.md` page for this new config option.

Reply:
I agree that the name should be `lagStatsType` or if you just want to use max, then `useMaxLag`. I don't mind it being an enum, my only concern is that it is limiting. What do we do if we want something outside of total/max/avg?

The approach in this PR is not exactly the same as the original PR, because here we drive the decision by a config and not a method on the `Supervisor` itself.

I would prefer it we could make the following changes in this PR:
1. If we intend to drive the decision via a config, let's get rid of the method `computeLagForAutoScaler()` altogether.
2. Update `LagStats` (returned by `computeLagStats()`) to have a field `preferredScalingMetric` which would be the new enum we add. For now, this value will be `TOTAL` for both Kinesis and Kafka. After we have done some validation, the `KinesisSupervisor.computeLagStats()` would return a `LagStats` with `preferredScalingMetric` as `MAX` for Kinesis.
3. The new config will essentially override the `preferredScalingMetric`. If the config is not specified, we use the `preferredScalingMetric` returned by the `computeLagStats` method. This logic should be in the `LagBasedAutoScaler` itself.
4. Let's have a plan on how we will eventually get rid of the new config or atleast compute its default value better. Because a single default value of this config is not going to work for all kinds of supervisors. We wouldn't want users to have to specify this config in every supervisor spec. I think step 2 above will help with this.

---

My concerns with the original approach were:
1. It was computing the lag in a roundabout manner. Supervisor needed to expose two methods `computeLagStats` and `getLagMetric`. Then the auto-scaler had to use the result of these two methods to determine the actual lag. If it is the Supervisor which drives this computation, the Supervisor might as well have a single method that returns the required lag value.
2. Enum is limited to TOTAL, MAX, AVG. Having a separate method for `computeLagForAutoScaler` allows us to keep the implementation more flexible. I don't know when we would need a value outside of total / max / avg though, perhaps if we wanted a normalized / truncated value of one of these metrics.
3. I didn't fully like the idea of adding a new method `computeLagForAutoScaler` either because the method `computeLagStats` itself was meant only to be used by the auto-scaler. Why have one more method? I suggested that approach as I didn't have a better alternative at that time which could move things along faster.

Reply:
@kfaraz Cool, makes sense to me.